tests/components/volvo/test_sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Test Volvo sensors."""
+
+from collections.abc import Awaitable, Callable
+from unittest.mock import patch
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import model
+
+
+@pytest.mark.parametrize(
+    ("keys"),
+    [
+        pytest.param(
+            [
+                "battery_charge_level",
+                "car_connection",
+                "charging_connection_status",
+                "distance_to_empty_battery",
+                "distance_to_service",
+                "engine_time_to_service",
+                "est_charging_time",
+                "odometer",
+                "time_to_service",
+                "tm_avg_energy_consumption",
+                "tm_avg_speed",
+                "tm_distance",
+            ],
+            marks=model("xc40_electric_2024"),
+        ),
+        pytest.param(
+            [
+                "car_connection",
+                "distance_to_empty_tank",
+                "distance_to_service",
+                "engine_time_to_service",
+                "fuel_amount",
+                "odometer",
+                "time_to_service",
+                "tm_avg_fuel_consumption",
+                "tm_avg_speed",
+                "tm_distance",
+            ],
+            marks=model("s90_diesel_2018"),
+        ),
+    ],
+)
+async def test_sensor(
```

Comment:
Use the `snapshot_platform` helper instead and just call it with the different models.